src/qtt/measurements/scans.py

```diff
@@ -1811,10 +1811,8 @@ def measuresegment(waveform, Naverage, minstrhandle, read_ch, mV_range=2000, pro
     if device_parameters is None:
         device_parameters = {}
         
-    try:
-        is_m4i = isinstance(minstrhandle, qcodes.instrument_drivers.Spectrum.M4i.M4i)
-    except:
-        is_m4i = False
+    is_m4i = _is_m4i(minstrhandle)
+
     try:
```

Comment:
if is_m4i is True it is not an uhfli or a scope reader. Why trying? The same for trying scope reader when minsthandle is a uhfli? A little bit of boyscouting needed here.

Reply:
Often these if elif elif else constructions depending on the type of instance can be refactored to a proper design (inheritance structure with an overridden method measure_segment). What is your opinion on these type of changes.